src/main/scala/mesosphere/marathon/core/storage/repository/impl/BaseAppRepository.scala

```diff
@@ -0,0 +1,47 @@
+package mesosphere.marathon.core.storage.repository.impl
+
+import java.time.OffsetDateTime
+
+import akka.http.scaladsl.marshalling.Marshaller
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.stream.scaladsl.Source
+import akka.{ Done, NotUsed }
+import mesosphere.marathon.core.storage.impl.memory.{ Identity, RamId }
+import mesosphere.marathon.core.storage.impl.zk.{ ZkId, ZkSerialized }
+import mesosphere.marathon.core.storage.repository.AppRepository
+import mesosphere.marathon.core.storage.{ IdResolver, PersistenceStore }
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+
+import mesosphere.marathon.core.storage.impl.memory.InMemoryStoreSerialization._
+import mesosphere.marathon.core.storage.impl.zk.ZkStoreSerialization._
+
+import scala.concurrent.Future
+
+abstract class BaseAppRepository[K, C, S](store: PersistenceStore[K, C, S], maxVersions: Int)(implicit
```

Comment:
See my comment above: all methods simply delegate to the persistence store, while resolving the implicits.